src/openrct2/network/ServerList.cpp

```diff
@@ -80,3 +233,170 @@ bool server_list_write(const std::vector<server_entry>& entries)
+        auto udpSocket = CreateUdpSocket();
+
+        log_verbose("Broadcasting %zu bytes to the LAN (%s)", msg.size(), broadcastAddress.c_str());
+        auto len = udpSocket->SendData(broadcastAddress, NETWORK_LAN_BROADCAST_PORT, msg.data(), msg.size());
+        if (len != msg.size())
+        {
+            throw std::runtime_error("Unable to broadcast server query.");
+        }
+
+        std::vector<ServerListEntry> entries;
+        char buffer[1024]{};
+        size_t recievedLen{};
+        std::unique_ptr<INetworkEndpoint> endpoint;
+        for (int i = 0; i < (RECV_WAIT_MS / RECV_DELAY_MS); i++)
+        {
+            auto p = udpSocket->ReceiveData(buffer, sizeof(buffer) - 1, &recievedLen, &endpoint);
+            if (p == NETWORK_READPACKET_SUCCESS)
+            {
+                auto sender = endpoint->GetHostname();
+                log_verbose("Received %zu bytes back from %s", recievedLen, sender.c_str());
+                auto jinfo = Json::FromString(std::string_view(buffer));
+
+                auto ip4 = json_array();
+                json_array_append_new(ip4, json_string(sender.c_str()));
+                auto ip = json_object();
+                json_object_set_new(ip, "v4", ip4);
+                json_object_set_new(jinfo, "ip", ip);
+
+                auto entry = ServerListEntry::FromJson(jinfo);
+                if (entry.has_value())
+                {
+                    (*entry).local = true;
+                    entries.push_back(*entry);
+                }
+
+                json_decref(jinfo);
+            }
+            platform_sleep(RECV_DELAY_MS);
+        }
+        return entries;
+    });
+}
+
+std::future<std::vector<ServerListEntry>> ServerList::FetchLocalServerListAsync()
+{
+    return std::async(std::launch::async, [&] {
+        // Get all possible LAN broadcast addresses
+        auto broadcastEndpoints = GetBroadcastAddresses();
+
+        // Spin off a fetch for each broadcast address
+        std::vector<std::future<std::vector<ServerListEntry>>> futures;
+        for (const auto& broadcastEndpoint : broadcastEndpoints)
+        {
+            auto f = FetchLocalServerListAsync(*broadcastEndpoint);
+            futures.push_back(std::move(f));
+        }
+
+        // Wait and merge all results
+        std::vector<ServerListEntry> mergedEntries;
+        for (auto& f : futures)
+        {
+            try
+            {
+                auto entries = f.get();
+                mergedEntries.insert(mergedEntries.begin(), entries.begin(), entries.end());
+            }
+            catch (...)
+            {
+                // Ignore any exceptions from a particular broadcast fetch
+            }
+        }
+        return mergedEntries;
+    });
+}
+
+std::future<std::vector<ServerListEntry>> ServerList::FetchOnlineServerListAsync()
+{
+#    ifdef DISABLE_HTTP
+    return std::async(std::launch::deferred, [] { return std::vector<ServerListEntry>(); });
```

Comment:
`std::launch::deferred` shouldn't spawn a thread... but I have changed it anyway. The `get()`s are already wrapped in try...catches.

Reply:
Sorry I actually misread the documentation, it does indeed not spawn a new thread, what I've misread is that it can be used by another thread and then invokes the function.